include/boost/geometry/arithmetic/general_form.hpp

```diff
@@ -0,0 +1,161 @@
+// Boost.Geometry
+
+// Copyright (c) 2018-2019 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ARITHMETIC_GENERAL_FORM_HPP
+#define BOOST_GEOMETRY_ARITHMETIC_GENERAL_FORM_HPP
+
+#include <boost/geometry/util/math.hpp>
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/config.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace arithmetic
+{
+
+//--------------------------------------------------------------------------
+// Structure containing the General Form of a line, a*x + b*y + c == 0
+// Might be conceptized later. Therefore operations are implemented outside
+// the structure itself.
+template <typename GeneralType = double>
+struct general_form
+{
+    general_form()
+        : a(GeneralType())
+        , b(GeneralType())
+        , c(GeneralType())
+        , normalized(false)
+    {}
+
+    // Horizontal: a == 0, for example y-3=0, y==3
+    // Vertical: b == 0, for example x-2=0, x==2
+    // Through origin: c == 0
+    GeneralType a;
+    GeneralType b;
+    GeneralType c;
+    bool normalized;
+};
+
+template <typename GeneralType, typename Coordinate>
+inline
+general_form<GeneralType> construct_line(Coordinate const& x1,
+    Coordinate const& y1, Coordinate const& x2, Coordinate const& y2)
```

Comment:
Is this the only kind of line which could be constructed from 4 numbers? This could either be the constructor of `general_line` type (or whatever other name it'd have) or this function could take the `general_line` as an argument in case you have more line types that could be constructed in the future.

Reply:
Most code we have is for 2d